app/src/main/java/com/example/bucketbuds/FriendsAdapter.java

```diff
@@ -0,0 +1,188 @@
+package com.example.bucketbuds;
+
+import android.content.Context;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.fragment.app.Fragment;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.bumptech.glide.Glide;
+import com.parse.FindCallback;
+import com.parse.ParseException;
+import com.parse.ParseUser;
+import com.parse.SaveCallback;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class FriendsAdapter extends RecyclerView.Adapter<FriendsAdapter.ViewHolder> {
+
+    public static final String TAG = "FriendsAdapter";
+
+    Context context;
+    List<User> users;
+    Boolean friendsBool;
+    Fragment fragment;
+    List<String> friendsId;
+    UserPub currentUserPub;
+    public static final User currentUser = User.getCurrentUser();
+
+    // pass in context and list of users
+    public FriendsAdapter(Context context, List<User> users, UserPub currentUserPub, Boolean friendsBool, Fragment fragment, List<String> friendsId){
+        this.friendsBool = friendsBool;
+        this.context = context;
+        this.users = users;
+        this.currentUserPub = currentUserPub;
+        this.fragment = fragment;
+        this.friendsId = friendsId;
+    }
+
+    @NonNull
+    @NotNull
+    @Override
+    public ViewHolder onCreateViewHolder(@NonNull @NotNull ViewGroup parent, int viewType) {
+        View view;
+        if (friendsBool) {
+            view = LayoutInflater.from(context).inflate(R.layout.item_friend, parent, false);
+        } else {
+            view = LayoutInflater.from(context).inflate(R.layout.item_add_friend, parent, false);
+        }
```

Comment:
nit: I think this makes sense, but a comment on what `friendsBool` means would be useful.